Michael/dragonswap plugin

MichaelDeng03 opened the pull request:
Add swap functions for dragonswap v2.

MichaelDeng03 committed:
Add exactInputSingle tool

MichaelDeng03 committed:
Add exactOutputSingle

MichaelDeng03 committed:
Remove extra factory address

MichaelDeng03 committed:
Remove extra imports

adgianv commented on src/tools/dragonswap/exactOutputSingle.ts:
```diff
@@ -0,0 +1,100 @@
+    functionName: "exactInputSingle",
+    args: [exactOutputSingleParams],
+  });
+}
```
```suggestion
return hashSwap;
}
```

adgianv commented on src/tools/dragonswap/exactOutputSingle.ts:
```diff
@@ -0,0 +1,100 @@
+  agent: SeiAgentKit,
+  tokenInAddress: Address,
+  tokenOutAddress: Address,
+  tokenInMaximum: bigint,
```
Is this any different from 'amountInMaximum'? 
If not delete it and modify the writeContact below, if yes add a docstring to explain the difference.

MichaelDeng03 replied:
Ope, no difference. fixed now

adgianv commented on src/tools/dragonswap/exactInputSingle.ts:
```diff
@@ -0,0 +1,105 @@
+    functionName: "exactInputSingle",
+    args: [exactInputSingleParams],
+  });
+}
```
```suggestion
return hashSwap;
}
```

adgianv commented on src/tools/dragonswap/index.ts:
```diff
@@ -0,0 +1 @@
+export * from "./exactInputSingle";
```
```suggestion
export * from "./exactInputSingle";
export * from "./exactOutputSingle";
```

MichaelDeng03 replied:
👍

adgianv commented on src/tools/dragonswap/exactInputSingle.ts:
```diff
@@ -0,0 +1,105 @@
+  const hashApprove = await walletClient.writeContract({
+    address: tokenInAddress,
+    abi: [
+      "function approve(address spender, uint256 amount) public returns (bool)",
```
This line was causing this error:
TypeError: Cannot use 'in' operator to search for 'name' in function approve(address spender, uint256 amount) public returns (bool)

I think you should have something like:
{
        type: "function",
        name: "approve",
        stateMutability: "nonpayable",
        inputs: [
          { name: "spender", type: "address" },
          { name: "amount", type: "uint256" }
        ],
        outputs: [{ type: "bool" }]
      }

MichaelDeng03 replied:
Ahh, I didn't realize there were such major differences between ethers and viem.

MichaelDeng03 committed:
Add dragonswap tools to agent.

MichaelDeng03 committed:
Use viem abi instead of ethers string abi.

MichaelDeng03 committed:
Add a missing export from dragonswap index.

MichaelDeng03 committed:
Init dragonswap tests and call it in primary test module

MichaelDeng03 committed:
Fix abi

MichaelDeng03 committed:
Remove try/catch in dragonswaptests and let main tests handle them

MichaelDeng03 committed:
Add awaits for transactions.

MichaelDeng03 commented:
Hey, I addressed some of your comments and bug fixed those abis.
I also added a few tests that are passing for me. Just make sure you have a starting balance of sei and wsei.
